File: Back/Pragmap/Pragmap.Domain/Entities/RoadMap.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Pragmap.Domain.Common.Interfaces;

namespace Pragmap.Domain.Entities
{
    public class Roadmap : IDatedEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        
        [MaxLength(255)]
        public string Name { get; set; }

        [Column(TypeName = "jsonb")]
        public string? Data { get; set; } 

        public Guid CustomerId { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
        public virtual Customer Customer { get; set; }
    }
}

File: Front/Pragmap/src/types/roadmap.ts
export interface IRoadmap {
	id: string
	name: string
	data: string
	customerId: string
	customerName: string
	createdAt: string
	updatedAt: string
}
export interface IGetRoadmap {
	id: string
	name: string
	data: string
	customerId: string
	createdAt: string
	updatedAt: string
}
export interface IPostRoadmap {
	name: string
	customerId: string
}

export interface IPutRoadmap {
	id: string
	name?: string
	data?: string
}
